### ISSUE TITLE
callback only called when app wasn't running

### ISSUE THREAD

**Issue by Burgov** [hypery2k/nativescript-urlhandler#69]
I have the following intent-filter:

```
		<activity
			android:name="com.tns.NativeScriptActivity"
			android:label="@string/title_activity_kimera"
			android:configChanges="keyboardHidden|orientation|screenSize"
			android:theme="@style/LaunchScreenTheme"
			android:launchMode="singleTask">

			<meta-data android:name="SET_THEME_ON_LAUNCH" android:resource="@style/AppTheme" />

			<intent-filter>
				<action android:name="android.intent.action.MAIN" />
				<category android:name="android.intent.category.LAUNCHER" />
			</intent-filter>

			<intent-filter>
				<data android:scheme="https" android:host="some.hostname.com" />
				<action android:name="android.intent.action.VIEW" />
				<category android:name="android.intent.category.DEFAULT" />
				<category android:name="android.intent.category.BROWSABLE" />
			</intent-filter>
		</activity>
```

and a `handleOpenURL(...)` call in the AppModule constructor. It works wonderfully, but only the first time a link is clicked, triggering the app to load up. If the app was already running when the link is clicked, it's brought to foreground, but the callback for handleOpenURL isn't called.

Am I missing something or is this a bug?

**Comment by hypery2k** [comment]
if it's the same url it would ignore a second call, that was by intention. So can you try to open another link when app is already open and check if this works?

**Comment by Burgov** [comment]
@hypery2k I saw in the code that it's indeed ignoring the url if it's the same as the last one, but why is that? Shouldn't that be sth for the handler to decided?

However, that doesn't seem to be the problem. My function never gets called if the app is running in BG, only when i completely close it. I tried it with 2 different links. Just brings the app to FG and that's it. Anything I can do to debug?

**Comment by maxfriedmann** [comment]
I debugged a bit, the intent during onResume is the wrong one, tried with all launchModes, see https://stackoverflow.com/a/6357330/1227985

I went for the solution with my own Activity/Delegate as described in https://github.com/ddfreiling/tns-ng-deeplink

**Comment by jdolderer** [comment]
I run into the same issue as @Burgov. Is there any solution how to fix this behavior?
On iOS the callback is working just fine, it happens only on Android.

**Comment by hypery2k** [comment]
Currently don't have to time to watch on this. Any PR is welcome

**Comment by hypery2k** [comment]
fixed in 1.2.3
